Content.Server/Paper/PaperSystem.cs

```diff
@@ -152,22 +179,22 @@ public void SetContent(EntityUid uid, string content, PaperComponent? paperComp
             paperComp.Content = content + '\n';
             UpdateUserInterface(uid, paperComp);
 
-            if (!TryComp<AppearanceComponent>(uid, out var appearance))
+            if (!HasComp<AppearanceComponent>(uid))
                 return;
 
-            var status = string.IsNullOrWhiteSpace(content)
-                ? PaperStatus.Blank
-                : PaperStatus.Written;
-
-            appearance.SetData(PaperVisuals.Status, status);
+            UpdateAppearance(uid, paperComp);
         }
 
         public void UpdateUserInterface(EntityUid uid, PaperComponent? paperComp = null)
         {
             if (!Resolve(uid, ref paperComp))
                 return;
 
-            _uiSystem.GetUiOrNull(uid, PaperUiKey.Key)?.SetState(new PaperBoundUserInterfaceState(paperComp.Content, paperComp.Mode));
+            var ui = _uiSystem.GetUiOrNull(uid, PaperUiKey.Key);
```

Comment:
don't get the ui if you're just going to use a TrySetUiState. you can just remove this and the if statement and leave the actual function call as is.